### DISCUSSION TITLE
Integrate with datatables JS

### DISCUSSION THREAD

**Discussion by JDavidVR** [breezefront/community#216] [1. Help]
Hi I'm new using Breezefront, so I would like to have a guidence on integrating it with third party JS (in this case it would be datatables)

In datatables https://datatables.net/download/ it generates following css and js files:

```
<link href="https://cdn.datatables.net/1.13.8/css/dataTables.jqueryui.css" rel="stylesheet">
<link href="https://cdn.datatables.net/responsive/2.5.0/css/responsive.jqueryui.css" rel="stylesheet">
 
<script src="https://cdn.datatables.net/1.13.8/js/jquery.dataTables.js"></script>
<script src="https://cdn.datatables.net/1.13.8/js/dataTables.jqueryui.js"></script>
<script src="https://cdn.datatables.net/responsive/2.5.0/js/dataTables.responsive.js"></script>
<script src="https://cdn.datatables.net/responsive/2.5.0/js/responsive.jqueryui.js"></script>
```

Should I integrate those js files as components in breeze_default.xml? (https://breezefront.com/docs/integration-example#3-register-js-components)

In luma I've integrated them in Vendor/Module/view/frontend/requirejs-config.js

```
var config = {
    map: {
        '*': {
            "datatables.net": "CustomerParadigm_DataTables/js/jquery.dataTables",
            "dtScroller": "CustomerParadigm_DataTables/js/dataTables.scroller"
        }
    },
    shim: {
        "datatables.net": ['jquery']
    }
}
```
then in phtml file: Vendor/Module/view/frontend/templates/datatable.phtml I can especify dependencies I need

```
<script>
require(['jquery', 'datatables.net', 'dtScroller'], function ($) {
    $(document).ready(function () {
        let table = $('#ajax').DataTable(configs);
    });
});
</script>
```

Thanks in advance!

**Comment by vovayatsyuk** [comment]
Yes, you have to register these js files - this is how we know that these files need to be added to the page. But I don't know if this plugin will work since we don't have jQuery.
